src/starlite_saqlalchemy/dto.py

```diff
@@ -242,43 +242,52 @@ class User(DeclarativeBase):
     exclude_ = exclude or set[str]()
     annotations_ = annotations or {}
     namespace_ = namespace or {}
-    validators_ = validators or {}
-    del exclude, annotations, namespace, validators
+    del exclude, annotations, namespace
     columns, relationships = _inspect_model(model)
     model_module = getmodule(model)
     localns = vars(model_module) if model_module is not None else {}
     localns.update(namespace_)
     type_hints = get_type_hints(model, localns=localns)
     type_hints.update(annotations_)
     fields: dict[str, tuple[Any, FieldInfo]] = {}
+    validators: dict[str, AnyClassMethod] = {}
     for key, type_hint in type_hints.items():
         if get_origin(type_hint) is Mapped:
             (type_,) = get_args(type_hint)
-        elif type_hint in annotations_:
-            type_ = type_hint
         else:
-            continue
+            type_ = type_hint
 
         elem: Column | RelationshipProperty
-        try:
+        if key in columns:
             elem = columns[key]
-        except KeyError:
+        elif key in relationships:
             elem = relationships[key]
+        else:
+            # class var, anything else??
+            continue
+
         attrib = _get_dto_attrib(elem)
+
         if _should_exclude_field(purpose, elem, exclude_, attrib):
             continue
 
+        if attrib.pydantic_type is not None:
+            type_ = attrib.pydantic_type
+
+        for i, func in enumerate(attrib.validators or []):
+            validators[f"_validates_{key}_{i}"] = validator(key, allow_reuse=True)(func)
+
         if isclass(type_) and issubclass(type_, DeclarativeBase):
             type_ = factory(f"{name}_{type_.__name__}", type_, purpose=purpose)
 
         fields[key] = (type_, _construct_field_info(elem, purpose))
 
     return create_model(  # type:ignore[no-any-return,call-overload]
         name,
-        __module__=getattr(model, "__module__", __name__),
         __base__=base,
```

Comment:
If `base` is explictly passed the produced does not inherit `MapperBind`. We should make sure that `MapperBind` is part of bases in all cases.

Reply:
Yah, this does need more thought.

The decorated class could inherit from `MapperBind` instead of `BaseModel`.

If the decorated class inherits from `BaseModel` then we can't simply mix in `MapperBind` as it inherits from `BaseModel` also.

If the decorated class inherits from `MapperBind` there is the issue that `model` needs to be passed as a class kwarg, so the model then gets declared in the decorator signature, and in the class signature on the very next line.. and I was unsure about the ergonomic of that.

I'll take another look with fresh eyes tomorrow.. I experimented with making `MapperBind` a mixin class, rather than inherit from `BaseModel` - maybe that's worth revisiting.

Reply:
Yes, I think `MapperBind` should be a mixin rather than a full pydantic model. Both the decorated class and `MapperBind` should be in bases, so we only want one of them to inherit `BaseModel`, or none.

Reply:
Got this where I'm pretty happy implementation-wise, does it feel close enough to what you were heading toward with the original PR? This test is a good demo:

```python
def test_dto_decorator() -> None:

    @dto.decorator(Author, dto.Purpose.WRITE)
    class AuthorDTO(BaseModel):
        name: constr(to_upper=True)  # pyright:ignore

        @validator("name")
        def validate_name(cls, val: str) -> str:
            """We're shouting!"""
            return f"{val}!"

        @validator("dob", check_fields=False)
        def validate_dob(cls, val: date) -> date:
            """Off by one."""
            val += timedelta(days=1)
            return val

    assert AuthorDTO.parse_obj({"name": "Bill Bryson", "dob": "1951-12-08"}).dict() == {
        "name": "BILL BRYSON!",
        "dob": date(1951, 12, 9),
    }
```

Still need to do docs, more tests, I think there'll be an issue with self-referencing relationships to sort out, and will probably break up the module into a sub-package as a bit going on in there now.. 

Main issue is that mypy isn't recognising the `to_mapped()` method, either via the `factory()` or `decorator()` method, just get these:

```
tests/utils/controllers.py:33: error: "CreateDTO" has no attribute "to_mapped"  [attr-defined]
tests/utils/controllers.py:45: error: UpdateDTO? has no attribute "to_mapped"  [attr-defined]
tests/unit/test_dto.py:253: error: "CreateDTO" has no attribute "to_mapped"  [attr-defined]
```

Seems like there is a very longstanding issue with mypy and class decorators, https://github.com/python/mypy/issues/3135#issuecomment-1311118319 but I wouldn't have thought that would affect the `factory()` generated DTOs.